Swift/Collection.swift

```diff
@@ -107,6 +107,9 @@ public final class Collection : CollectionChangeObservable, Indexable, Equatable
     /// Throws an NSError with the CBLError.notOpen code, if the collection is deleted or
     /// the database is closed.
     public func save(document: MutableDocument) throws {
+        if (document.collection == nil){
+            document.collection = self
+        }
```

Comment:
This should be done after saving is successful.

Reply:
There are another two save functions that need set the collection to the saved doc as well.